### PR TITLE
fix(abap-deploy-config-writer): add `reentranceTicket` auth type to all deploy configs when passed in

### PR DESCRIPTION
**Pull request by cianmSAP**
#2555 

No longer relies on reading the base config for `reentranceTicket` auth type, it is passed in with the config param
Removes the check for AppStudio when writing `authenticationType: reentranceTicket`
Also adds the comment `# SAML support for vscode` to align with other config files

### CODE REVIEW

File: packages/abap-deploy-config-writer/src/config.ts [hunk 1]
```diff
@@ -47,13 +48,17 @@ export async function getDeployConfig(config: AbapDeployConfig, baseConfig: UI5C
...
-            }
+    if (config.target.authenticationType === 'reentranceTicket') {
+        target.authenticationType = 'reentranceTicket';
+        if (target.url) {
+            target.url = removeApiHostname(target.url);
         }
+
+        comments.push({
+            path: 'configuration.target.authenticationType',
+            comment: ' SAML support for vscode',
```
Comment (longieirl) [inline packages/abap-deploy-config-writer/src/config.ts:59]: Do you need the space in the comment?

Comment (longieirl) [reply packages/abap-deploy-config-writer/src/config.ts:59]: Yes you do as I saw your next change in the YAML file! Thanks.

File: packages/fiori-generator-shared/src/helpers.ts [hunk 2]
```diff
@@ -58,3 +58,15 @@ export function getDefaultTargetFolder(vscode: any): string | undefined {
     // Otherwise use <home-dir>/projects,
     return existsSync(DEFAULT_PROJECTS_FOLDER) ? DEFAULT_PROJECTS_FOLDER : undefined;
 }
+
+/**
+ * Removes any `-api` suffix in the first label of the hostname.
```
Comment (longieirl) [inline packages/fiori-generator-shared/src/helpers.ts:63]: You might add the `why` we are adding this method? For example, an external system knows it should connect to https://<tenant-name>-api.s4hana.ondemand.com for API calls, as opposed to interacting with the standard UI hostname, https://<tenant-name>.s4hana.ondemand.com.

Comment (cianmSAP) [reply packages/fiori-generator-shared/src/helpers.ts:63]: Comment added

File: packages/abap-deploy-config-writer/src/config.ts [hunk 3]
```diff
@@ -47,13 +48,17 @@ export async function getDeployConfig(config: AbapDeployConfig, baseConfig: UI5C
...
-    if (!isAppStudio()) {
-        const backendConfigs = baseConfig.getBackendConfigsFromFioriToolsProxydMiddleware();
-        for (const backend of backendConfigs) {
-            if (backend.authenticationType === 'reentranceTicket') {
-                target.authenticationType = 'reentranceTicket';
-            }
+    if (config.target.authenticationType === 'reentranceTicket') {
+        target.authenticationType = 'reentranceTicket';
+        if (target.url) {
+            target.url = removeApiHostname(target.url);
```
Comment (IainSAP) [inline packages/abap-deploy-config-writer/src/config.ts:54]: Do we want to do this in VSCode?

**Review by IainSAP — COMMENTED** [review]
Could you please address (the url modification) in the  inquirer also? And log.

**Comment by cianmSAP** [comment]
> Could you please address (the url modification) in the inquirer also? And log. The concern is we allow a bad url in and it may be written into other places including the `-api`. If this is needed urgently then lets create a follow-up tbi?

I'm going to isolate these PRs 

This PR is just to add the auth type (with comment) to the deploy.yaml

**Review by IainSAP — APPROVED** [review]
Thx @cianmSAP 

- Changes are clear
- Tests updated to cover

**Review by Klaus-Keller — APPROVED** [review]
Thanks @cianmSAP, 
* code changes for authentication type are straight forward and easy to understand
* changeset exists
* test coverage is great
* did a visual review, no manual test

Approved from my side.

**Pull request merged**